app/root.tsx

```diff
@@ -1,11 +1,19 @@
-import type { LinksFunction } from '@remix-run/node';
+import type { LinksFunction, LoaderArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
 import stylesheet from './styles/tailwind.css';
 import TopNav from '~/components/top-nav';
 import { Links, LiveReload, Meta, Outlet, Scripts, ScrollRestoration } from '@remix-run/react';
 import Footer from './components/footer';
+import { getUserSession } from '~/modules/session/session.server';
 
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: stylesheet }];
 
+export async function loader({ request }: LoaderArgs) {
+  const session = await getUserSession(request);
```

Comment:
NIT: Maybe we can use the `session.userId` to query the user from the database and return the user instead? Could be useful to display the user profile picture and display name.

Reply:
Returning the user instead sounds like a great idea. I'll make the updates.